frontend/controller/actions/identity.js

```diff
@@ -701,5 +701,72 @@ export default (sbp('sbp/selectors/register', {
   ...encryptedAction('gi.actions/identity/joinDirectMessage', L('Failed to join a direct message.')),
   ...encryptedAction('gi.actions/identity/joinGroup', L('Failed to join a group.')),
   ...encryptedAction('gi.actions/identity/leaveGroup', L('Failed to leave a group.')),
-  ...encryptedAction('gi.actions/identity/setDirectMessageVisibility', L('Failed to set direct message visibility.'))
+  ...encryptedAction('gi.actions/identity/setDirectMessageVisibility', L('Failed to set direct message visibility.')),
+  'gi.actions/identity/uploadFiles': async ({ attachments, billableContractID }: {
+    attachments: Object, billableContractID: string
+  }) => {
+    const rootGetters = sbp('state/vuex/getters')
+
+    try {
+      const attachmentsData = await Promise.all(attachments.map(async (attachment) => {
+        const { mimeType, url } = attachment
+        // url here is an instance of URL.createObjectURL(), which needs to be converted to a 'Blob'
+        const attachmentBlob = await objectURLtoBlob(url)
+        const response = await sbp('chelonia/fileUpload', attachmentBlob, {
+          type: mimeType, cipher: 'aes256gcm'
+        }, { billableContractID })
+        const { delete: token, download: downloadData } = response
+        return {
+          downloadData: { ...omit(attachment, ['url']), downloadData },
+          deleteData: { token }
+        }
+      }))
+
+      const tokensByManifestCid = attachmentsData.map(({ downloadData, deleteData }) => ({
+        manifestCid: downloadData.downloadData.manifestCid,
+        token: deleteData.token
+      }))
+
+      await sbp('gi.actions/identity/saveFileDeleteToken', {
+        contractID: rootGetters.ourIdentityContractId,
+        data: { tokensByManifestCid }
+      })
+
+      return attachmentsData.map(({ downloadData }) => downloadData)
+    } catch (err) {
+      console.error('Error during uploading files', err)
+    }
+  },
+  'gi.actions/identity/removeFiles': async ({ manifestCids, identityContractID }: {
+    manifestCids: string[], identityContractID: string, billableContractID?: string
+  }) => {
+    const rootGetters = sbp('state/vuex/getters')
+    const me = sbp('state/vuex/state').loggedIn.identityContractID
+
+    if (me !== identityContractID) {
+      return
```

Comment:
Probably a good idea to issue a `console.warn` here before returning

Reply:
At the moment, we don't have any user flow for the group creators to delete attachments. So I would like to consider this later when we have the design.

Or, should we simply make the `Delete` button visible for the attachments when group creator hovers on them?

Reply:
> should we simply make the Delete button visible for the attachments when group creator hovers on them? And one question related to this is that should it be possible for the group creators to delete messages sent by another?

Yes to both questions

Reply:
I will update the PR to make it possible for the group creators to delete attachments with the same user flow for the sender to delete them.
And for the second problem (It's actually a new workflow which makes me surprise), I will create another issue and implement it in another PR. Does that sounds good?

Reply:
Yep!